### PR TITLE
Added gpg sign commit for flyte-bot

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added gpg sign for flyte-bot commit 
- Added flyteidl automation 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: .github/workflows/boilerplate-automation.yml [hunk 1]
```diff
@@ -10,6 +10,12 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: "0"
+      - uses: crazy-max/ghaction-import-gpg@v3
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git-user-signingkey: true
```
Comment (evalsocket) [inline .github/workflows/boilerplate-automation.yml:17]: @EngHabu can you verify the secret name ?

File: .github/workflows/release-automation.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: crazy-max/ghaction-import-gpg@v3
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git-user-signingkey: true
+          git-commit-gpgsign: true
+      - name: Update release
```
Comment (pmahindrakar-oss) [inline .github/workflows/release-automation.yml:23]: Update FlyteIDL version

**Review by pmahindrakar-oss — COMMENTED** [review]
Can we remove 
`Unchanged files with check annotations`
as they create unnecessary noise in the PR

File: .github/workflows/release-automation.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: Update flyteidl version
```
Comment (pmahindrakar-oss) [inline .github/workflows/release-automation.yml:1]: Lets add  a section to the Contributing guide ,on how to test github workflow changes

**Review by pmahindrakar-oss — DISMISSED** [review]

File: .github/workflows/release-automation.yml (hunk 3 above)
Comment (evalsocket) [reply .github/workflows/release-automation.yml:1]: should we add this? Currently, we only have act but I am not sure act will work for everyone. i do my testing in my private repo but I will not recommend that

**Comment by evalsocket** [comment]
@pmahindrakar-oss I signed my commit. You need to review it again

File: .github/workflows/boilerplate-automation.yml [hunk 4]
```diff
@@ -10,6 +10,12 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: "0"
+      - uses: crazy-max/ghaction-import-gpg@v3
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
```
Comment (EngHabu) [inline .github/workflows/boilerplate-automation.yml:16]:
```suggestion
          passphrase: ${{ secrets.FLYTE_BOT_GPG_PASSPHRASE }}
```

File: .github/workflows/boilerplate-automation.yml [hunk 5]
```diff
@@ -10,6 +10,12 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: "0"
+      - uses: crazy-max/ghaction-import-gpg@v3
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
```
Comment (EngHabu) [inline .github/workflows/boilerplate-automation.yml:15]:
```suggestion
          gpg-private-key: ${{ secrets.FLYTE_BOT_GPG_PRIVATE_KEY }}
```

**Review by EngHabu — APPROVED** [review]

File: .github/workflows/release-automation.yml (hunk 3 above)
Comment (pmahindrakar-oss) [reply .github/workflows/release-automation.yml:1]: @evalsocket  would be good if we can add current approach for the community to contribute to changes in these actions. 
Lets add what you do for testing it right now and iterate and add the recommended approach as we find the best practices.

**Pull request merged**